### PR TITLE
implements user ssh key provider that respect RBAC policies

### LINKED ISSUE #1182: Implement RBAC compliant SSHKeyProvider
child of #966

### PR DESCRIPTION
**Pull request by p0lyn0mial**
**What this PR does / why we need it**: implements user ssh key provider that respect RBAC policies

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1182 

**Special notes for your reviewer**: the code is untested but I decided to share it anyway. I'll add more UTs in the future.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/pkg/provider/kubernetes/rbac-compliant-ssh.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+// ListOptions allows to set filters that will be applied
+// to filter the result of List method.
+type ListOptions struct {
+	// ClusterName gets the keys that are being used by the given cluster name
+	ClusterName string
+}
+
+// List gets a list of ssh keys, by default it will get all the keys that belong to the given project.
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/rbac-compliant-ssh.go:95]: note:
keys belong to a project not to a user
keys can be assigned/used by a cluster or clusters

File: api/pkg/provider/kubernetes/rbac-compliant-ssh.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+			Name:        keyName,
+			Clusters:    []string{},
+		},
+	}
+
+	masterImpersonatedClient, err := p.createMasterImpersonationClientWrapper(user, project)
+	if err != nil {
+		return nil, err
+	}
+	return masterImpersonatedClient.UserSSHKeies().Create(sshKey)
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/rbac-compliant-ssh.go:85]: could we also change the name `UserSSHKeies` to `SSHKeys` in a follow up ?

Comment (mrIncompetent) [reply api/pkg/provider/kubernetes/rbac-compliant-ssh.go:85]: I don't think thats possible. The codegen from kubernetes is responsible for this.
I tried changing it upstream. Was very complicated & caused a lot of changes upstream.
So i would recomment to stick with it

**Review by mrIncompetent — DISMISSED** [review]

File: api/pkg/provider/kubernetes/rbac-compliant-ssh.go (hunk 2 above)
Comment (p0lyn0mial) [reply api/pkg/provider/kubernetes/rbac-compliant-ssh.go:85]: oh, interesting, it sounds like an issue with plural form of nouns.

Comment (mrIncompetent) [reply api/pkg/provider/kubernetes/rbac-compliant-ssh.go:85]: In fact, it is a issue with the plural of nouns

**Comment by p0lyn0mial** [comment]
@mrIncompetent  just fixed the linter, tag the pull.

**Review by mrIncompetent — APPROVED** [review]

**Pull request merged**